src/Publish/Microsoft.NETCore.Sdk.Publish.Tasks/Tasks/TransformWebConfig.cs

```diff
@@ -74,6 +82,11 @@ public override bool Execute()
             string outputFile = Path.GetFileName(TargetPath);
             XDocument transformedConfig = WebConfigTransform.Transform(webConfigXml, outputFile, IsAzure, IsPortable);
 
+            if (string.IsNullOrEmpty(ProjectGuid) && !IgnoreProjectGuid)
+            {
+                ProjectGuid = WebConfigTransform.GetProjectGuidFromSolutionFile(SolutionPath, ProjectFullPath);
+            }
+
             // Add the projectGuid to web.config if it is not present.
             transformedConfig = WebConfigTransform.AddProjectGuidToWebConfig(transformedConfig, ProjectGuid, IgnoreProjectGuid);
```

Comment:
Yes, that's correct. I see now that there is no way for you to know at the time of publishing if telemetry is disabled on the server where the app will be deployed. Therefore, *you must default to adding the GUID here.* Anyone not wanting it does need to manually add the `IgnoreProjectGuid` property. :+1:

> check to make sure that if this is set, then "dotnet publish" wont track the projectGuid.

No need ... I've seen how they do it. It's good.

Thanks for your help. :smile: